cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsPlatformResources.java

```diff
@@ -771,31 +773,32 @@ private float getMemory(InstanceTypeInfo instanceType) {
     @Override
     public CloudGateWays gateways(CloudCredential cloudCredential, Region region, Map<String, String> filters) {
         Map<String, Set<CloudGateWay>> resultCloudGateWayMap = new HashMap<>();
-        CloudRegions regions = regions(cloudCredential, region, filters, true);
-
-        for (Entry<Region, List<AvailabilityZone>> regionListEntry : regions.getCloudRegions().entrySet()) {
-            if (region == null || Strings.isNullOrEmpty(region.value()) || regionListEntry.getKey().value().equals(region.value())) {
-                AmazonEc2Client ec2Client = awsClient.createEc2Client(new AwsCredentialView(cloudCredential), regionListEntry.getKey().value());
-
-                DescribeInternetGatewaysRequest describeInternetGatewaysRequest = new DescribeInternetGatewaysRequest();
-                DescribeInternetGatewaysResult describeInternetGatewaysResult = ec2Client.describeInternetGateways(describeInternetGatewaysRequest);
-
-                Set<CloudGateWay> gateWays = new HashSet<>();
-                for (InternetGateway internetGateway : describeInternetGatewaysResult.getInternetGateways()) {
-                    CloudGateWay cloudGateWay = new CloudGateWay();
-                    cloudGateWay.setId(internetGateway.getInternetGatewayId());
-                    cloudGateWay.setName(internetGateway.getInternetGatewayId());
-                    Collection<String> vpcs = new ArrayList<>();
-                    for (InternetGatewayAttachment internetGatewayAttachment : internetGateway.getAttachments()) {
-                        vpcs.add(internetGatewayAttachment.getVpcId());
+        if (region != null && !Strings.isNullOrEmpty(region.value())) {
```

Comment:
is that ok if the` resultCloudGateWayMap` empty

Reply:
that mean the region does not exist